Sources/CoreFoundation/BlockRuntime/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+# Build the vendored version of the BlocksRuntime library, which is used by
+# platforms that don't support libdispatch.
+
+add_library(BlocksRuntime
+    data.c
+    runtime.c)
+
+target_include_directories(BlocksRuntime PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}/include
+    # For CFTargetConditionals.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include)
+
+set_target_properties(BlocksRuntime PROPERTIES
+    POSITION_INDEPENDENT_CODE FALSE)
+
+add_library(BlocksRuntime::BlocksRuntime ALIAS BlocksRuntime)
+
+if(NOT BUILD_SHARED_LIBS)
+    set_property(GLOBAL APPEND PROPERTY Foundation_EXPORTS BlocksRuntime)
+    install(TARGETS BlocksRuntime
```

Comment:
Does WASI have a static SDK? If so, unfortunately just installing the library isn't sufficient. We need to ensure that it is auto linked from every swift module that imports the library. See the `autolink` flags added to Foundation/FoundationNetworking/FoundationXML for an example of this. That way the linker will link this library even though there isn't a public import of the module

Reply:
> Does WASI have a static SDK?
We don't support shared libraries but only support static libraries on WASI.

Given that BlocksRuntime is a private dependency of CoreFoundation, I think the best way is to build BlocksRuntime as object library and include the object files in libCoreFoundation.a. In this way, we don't need to worry about how to place it in the toolchain and autolinking stuff.